packages/taco/src/conditions/base/shared.ts

```diff
@@ -1,11 +1,19 @@
 import { z } from 'zod';
 
 import { ETH_ADDRESS_REGEXP, USER_ADDRESS_PARAM } from '../const';
+import { CONTEXT_PARAM_PREFIX } from '../context/context';
+
+export const ContextParamSchema = z.string().startsWith(CONTEXT_PARAM_PREFIX);
+export const ParamSchema = z.union([z.number(), z.string(), z.boolean()]);
+export const ParamOrContextParamSchema = z.union([
+  ParamSchema,
+  ContextParamSchema,
+]);
 
 export const returnValueTestSchema = z.object({
   index: z.number().optional(),
   comparator: z.enum(['==', '>', '<', '>=', '<=', '!=']),
-  value: z.unknown(),
+  value: ParamOrContextParamSchema,
```

Comment:
I don't think that `ParamSchema` defined as:
```typescript
export const ParamSchema = z.union([z.number(), z.string(), z.boolean()]);
```
covers everything here. I only gave `bool` as a single example in my comment. But truthfully, couldn't the value here be an array (of a variety of types). Basically, the example of a contract returning a struct (which is returned as tuple/list). See examples in `nucypher`.

While it is good to be as prescriptive as possible, we have to be careful about being too narrow to prevent some use cases which are still in the realm of solidity specification. I think this was the original reason why `z.unknown()` was used for a value.  In `nucypher` we use `fields.Raw` for `value` the same reason.